### PR TITLE
New makefile and BeWake.rdef

### PR DESCRIPTION
**Pull request by abt2804**
- Added BeWake.rdef and removed BeWake.rsrc
- Updated makefile

### CODE REVIEW

File: Source/BeWake.rdef [hunk 1]
```diff
@@ -0,0 +1,70 @@
+
```
Comment (humdingerb) [inline Source/BeWake.rdef:2]: Remove empty first line.

File: Source/BeWake.rdef [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+resource file_types message;
+
+resource large_icon array {
```
Comment (humdingerb) [inline Source/BeWake.rdef:18]: Put the icon data at the end of the file.

File: Source/BeWake.rdef [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	$"FF000A1C00000000173F3F113F3F00FF"
+	$"FF00001C1C00000000123F113F00FFFF"
+	$"FF000016161C000000003F3F000611FF"
+	$"FFFFFF000016161C1B000A000F061111"
+	$"FFFFFFFFFF000016151B1C150F061111"
+	$"FFFFFFFFFFFFFF06001516150F0011FF"
+	$"FFFFFFFFFFFFFFFFFF0600060011FFFF"
+};
+
+resource app_signature "application/x-vnd.simbad-BeWake";
```
Comment (humdingerb) [inline Source/BeWake.rdef]: Move all below to the top of the file.

File: Source/BeWake.rdef [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+resource app_version {
+	major  = 1,
+	middle = 1,
+	minor  = 0,
+
+	variety = B_APPV_DEVELOPMENT,
+	internal = 0,
+
+	short_info = "BeWake",
+	long_info = "BeWake screensaver module, v:1.1"
```
Comment (humdingerb) [inline Source/BeWake.rdef]: Let's remove the version ", v:1.1" from the long_info. That's what the "app_version" above is for.

File: Source/makefile [hunk 5]
```diff
@@ -30,23 +30,29 @@ TYPE= SHARED
 #	if two source files with the same name (source.c or source.cpp)
 #	are included from different directories.  Also note that spaces
 #	in folder names do not work well with this makefile.
-SRCS= BeWake.cpp	\
-	ConfigView.cpp	\
-	HueSlider.cpp	\
-	Utils.cpp
+SRCS =	\
```
Comment (humdingerb) [inline Source/makefile]: Please use a current makefile_engine makefile (In Tracker > New > Makefile).
In front of every "\" just one space.

File: Source/makefile [hunk 6]
```diff
@@ -30,23 +30,29 @@ TYPE= SHARED
...
+	 BeWake.cpp 	\
+	 ConfigView.cpp 	\
+	 HueSlider.cpp	\
+	 Utils.cpp	\
```
Comment (humdingerb) [inline Source/makefile]: No \ + linebreak after the last file.

File: Source/makefile [hunk 7]
```diff
@@ -30,23 +30,29 @@ TYPE= SHARED
...
+
 
 #	specify the resource definition files to use
 #	full path or a relative path to the resource file can be used.
-RDEFS= 
+RDEFS =	\
+                 BeWake.rdef	\
```
Comment (humdingerb) [inline Source/makefile]: No need to put it in an extra line, just put the single file right after the "RDEF =" above.

File: Source/makefile [hunk 8]
```diff
@@ -30,23 +30,29 @@ TYPE= SHARED
...
 	
 #	specify the resource files to use. 
 #	full path or a relative path to the resource file can be used.
 #	both RDEFS and RSRCS can be defined in the same makefile.
-RSRCS= BeWake.rsrc
+RSRCS = 
+
```
Comment (humdingerb) [inline Source/makefile]: No extra newline.

File: Source/makefile [hunk 9]
```diff
@@ -30,23 +30,29 @@ TYPE= SHARED
...
 
 # @<-src@ 
 #%}
 
+#%}
+
```
Comment (humdingerb) [inline Source/makefile:51]: No extra newline.

**Review by humdingerb — CHANGES_REQUESTED** [review]

File: Source/BeWake.rdef (hunk 2 above)
Comment (humdingerb) [reply Source/BeWake.rdef:18]: There's still app_signature and file_types down there which should be on top.

File: Source/BeWake.rdef [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
```
Comment (humdingerb) [inline Source/BeWake.rdef]: Remove ", v:1.1"...

File: Source/makefile (hunk 5 above)
Comment (humdingerb) [reply Source/makefile]: This is still an old makefile.

File: Source/makefile [hunk 11]
```diff
@@ -30,23 +30,26 @@ TYPE= SHARED
...
+SRCS =	\
+	 BeWake.cpp 	\
+	 ConfigView.cpp 	\
+	 HueSlider.cpp	\
+	 Utils.cpp
 
 #	specify the resource definition files to use
 #	full path or a relative path to the resource file can be used.
-RDEFS= 
+RDEFS =	BeWake.rdef
```
Comment (humdingerb) [inline Source/makefile]: This should be a space, not a tab.

File: Source/makefile [hunk 12]
```diff
@@ -30,23 +30,26 @@ TYPE= SHARED
...
```
Comment (humdingerb) [inline Source/makefile]: Remove the tab.

**Review by humdingerb — CHANGES_REQUESTED** [review]

**Review by humdingerb — CHANGES_REQUESTED** [review]
The RDEF is OK now. For the Makefile there are still open comments.

File: Source/makefile (hunk 5 above)
Comment (abt2804) [reply Source/makefile]: @humdingerb Where is that makefile?

Comment (humdingerb) [reply Source/makefile]: I wrote that in the comment above my above comment...

Comment (abt2804) [reply Source/makefile]: @humdingerb But I didn't find that file.

Comment (humdingerb) [reply Source/makefile]: The new "Makefile" appears where you invoked Tracker's "New > Makefile". If you right-clicked on the Desktop to do that, the Makefile appears on the Desktop.

Comment (abt2804) [reply Source/makefile]: I mean that I didn't find the option of "new"

Comment (humdingerb) [reply Source/makefile]: Right-click the Desktop. It's the second item in the context menu.

Comment (abt2804) [reply Source/makefile]: @humdingerb I am getting this problem when switched to new makefile - 
```mkdir -p objects.x86-cc2-release; \
mkdepend  -I./  -Iobjects.x86-cc2-release/ -p .cpp:objects.x86-cc2-release/%n.o -m -f "objects.x86-cc2-release/Utils.d" Utils.cpp
mkdir -p objects.x86-cc2-release; \
mkdepend  -I./  -Iobjects.x86-cc2-release/ -p .cpp:objects.x86-cc2-release/%n.o -m -f "objects.x86-cc2-release/HueSlider.d" HueSlider.cpp
mkdir -p objects.x86-cc2-release; \
mkdepend  -I./  -Iobjects.x86-cc2-release/ -p .cpp:objects.x86-cc2-release/%n.o -m -f "objects.x86-cc2-release/ConfigView.d" ConfigView.cpp
mkdir -p objects.x86-cc2-release; \
mkdepend  -I./  -Iobjects.x86-cc2-release/ -p .cpp:objects.x86-cc2-release/%n.o -m -f "objects.x86-cc2-release/BeWake.d" BeWake.cpp
g++ -c BeWake.cpp  -I./  -Iobjects.x86-cc2-release/ -I-   -O3    -o "objects.x86-cc2-release/BeWake.o"
g++ -c ConfigView.cpp  -I./  -Iobjects.x86-cc2-release/ -I-   -O3    -o "objects.x86-cc2-release/ConfigView.o"
g++ -c HueSlider.cpp  -I./  -Iobjects.x86-cc2-release/ -I-   -O3    -o "objects.x86-cc2-release/HueSlider.o"
g++ -c Utils.cpp  -I./  -Iobjects.x86-cc2-release/ -I-   -O3    -o "objects.x86-cc2-release/Utils.o"
cat BeWake.rdef | cc -E  -I./  -Iobjects.x86-cc2-release/ -I-   -O3    - | grep -av '^#' | rc -I ./ -o "objects.x86-cc2-release/BeWake.rsrc" -
cc -o "objects.x86-cc2-release/BeWake"  objects.x86-cc2-release/BeWake.o objects.x86-cc2-release/ConfigView.o objects.x86-cc2-release/HueSlider.o objects.x86-cc2-release/Utils.o   -L./  -Lobjects.x86-cc2-release/    -lbe  -lscreensaver 
/boot/system/develop/lib/start_dyn.o: In function `_start':
start_dyn.c:(.text+0x57): undefined reference to `main'
collect2: ld returned 1 exit status
/boot/system/develop/etc/makefile-engine:246: recipe for target 'objects.x86-cc2-release/BeWake' failed
make: *** [objects.x86-cc2-release/BeWake] Error 1```

Comment (abt2804) [reply Source/makefile]: Wait, it is fixed.

**Review by humdingerb — APPROVED** [review]

**Comment by humdingerb** [comment]
Thanks!

**Pull request merged**